### PR TITLE
compiler: Patch data dependencies across Jumps

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: devito/ir/support/basic.py [hunk 1]
```diff
@@ -822,7 +823,10 @@ def __init__(self, exprs, rules=None):
                     f = a.function
                     indices = [i if d in e.ispace else S.Infinity
                                for i, d in zip(a, a.aindices)]
-                    mock = f.indexify(indices)
+                    try:
+                        mock = f.indexify(indices)
+                    except IndexException:
```
Comment (mloubout) [inline devito/ir/support/basic.py:828]: What would make this fail? Bad indices?

Comment (FabioLuporini) [reply devito/ir/support/basic.py:828]: yeah just empty `indices`, trivial sympy exception when you attempt to create an Indexed w/o supplying indices 😬

Comment (georgebisbas) [reply devito/ir/support/basic.py:828]: Should you add some related message or failing test?

**Review by georgebisbas — APPROVED** [review]

**Review by mloubout — APPROVED** [review]

File: devito/ir/support/basic.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/ir/support/basic.py:828]: Definitely not a message; probably a unit test...

However, creating an artificial case exciting this code path here w/o PRO will take O(hour) time, which I don't have, so given that this is a minuscule thing and that, after all, we do have a test exciting this in PRO, despite agreeing w/ you this ain't ideal, I suggest we proceed

**Pull request merged**